### PR TITLE
fix(dj_search): :adhesive_bandage: fix additional error not being caught

### PR DESCRIPTION
**Pull request by iamamutt**
Was getting integrityerror instead of operationalerror

### CODE REVIEW

File: datajoint_utilities/dj_search/lists.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def drop_schemas(prefix, dry_run=True, ordered=False, force_drop=False):
             for schema_name in schemas_with_prefix:
                 try:
                     dj.schema(schema_name).drop(force=force_drop)
-                except OperationalError as e:
+                except (OperationalError, IntegrityError) as e:
```
Comment (dimitri-yatsenko) [inline datajoint_utilities/dj_search/lists.py:82]: There is a slight logical error with this code. The exception saved in `recent_err` may not be the most relevant exception that causes the failure to drop the entire group of schemas. The most informative error would be a foreign key constraint that references a table that is not included in the list, for example. It's fine for now. Perhaps listing all the errors would address the problem.

File: datajoint_utilities/dj_search/lists.py [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import datajoint as dj
-from pymysql import OperationalError
+from pymysql import IntegrityError, OperationalError
 
 """
 Development helper functions. Chris Brozdowski <CBroz@datajoint.com>
```
Comment (dimitri-yatsenko) [inline datajoint_utilities/dj_search/lists.py]: we don't need to put the authors in the code. The code version control system keeps track of everyone's contributions.

File: datajoint_utilities/dj_search/lists.py [hunk 3]
```diff
@@ -79,7 +79,7 @@ def drop_schemas(prefix, dry_run=True, ordered=False, force_drop=False):
             for schema_name in schemas_with_prefix:
                 try:
                     dj.schema(schema_name).drop(force=force_drop)
```
Comment (dimitri-yatsenko) [inline datajoint_utilities/dj_search/lists.py:81]: We need to modify `drop` to make it raise DataJoint errors rather than pymysql errors.

Comment (CBroz1) [reply datajoint_utilities/dj_search/lists.py:81]: I opened an [issue in datajoint-python](https://github.com/datajoint/datajoint-python/issues/1032) to this effect

File: datajoint_utilities/dj_search/lists.py (hunk 2 above)
Comment (CBroz1) [reply datajoint_utilities/dj_search/lists.py]: Sure, I can remove this in a PR to @iamamutt's fork [here](https://github.com/iamamutt/datajoint-utilities/pull/1)

Comment (iamamutt) [reply datajoint_utilities/dj_search/lists.py]: @CBroz1 @dimitri-yatsenko merged PR

**Review by CBroz1 — APPROVED** [review]

File: datajoint_utilities/dj_search/lists.py [hunk 4]
```diff
@@ -75,18 +74,20 @@ def drop_schemas(prefix, dry_run=True, ordered=False, force_drop=False):
...
         while schemas_with_prefix:
+            recent_errs = []  # Refresh recent_errs at start of while loop
             n_schemas_initial = len(schemas_with_prefix)
             for schema_name in schemas_with_prefix:
                 try:
                     dj.schema(schema_name).drop(force=force_drop)
-                except OperationalError as e:
-                    recent_err = e
+                except (OperationalError, IntegrityError) as e:
+                    recent_errs = [*recent_errs, str(e)]  # Add to list for current loop
```
Comment (dimitri-yatsenko) [inline datajoint_utilities/dj_search/lists.py]:
```suggestion
                    recent_errs.append(str(e))  # Add to list for current loop
```

File: datajoint_utilities/dj_search/lists.py [hunk 5]
```diff
@@ -35,8 +34,8 @@ def list_drop_order(prefix):
     while len(depends_on):
         drop_list += [k for k, v in depends_on.items() if not v]  # empty is dropable
         depends_on = {k: v for k, v in depends_on.items() if v}  # remove from dict
-        remaining = depends_on.keys()  # can't change dict in loop
-        for schema in remaining:  # remove dropable from other values
+        for schema in depends_on.keys():
```
Comment (dimitri-yatsenko) [inline datajoint_utilities/dj_search/lists.py]:
```suggestion
        for schema in depends_on:
```

File: datajoint_utilities/dj_search/lists.py [hunk 6]
```diff
@@ -35,8 +34,8 @@ def list_drop_order(prefix):
     while len(depends_on):
```
Comment (dimitri-yatsenko) [inline datajoint_utilities/dj_search/lists.py]:
```suggestion
    while depends_on:
```

File: datajoint_utilities/dj_search/lists.py [hunk 7]
```diff
@@ -75,18 +74,20 @@ def drop_schemas(prefix, dry_run=True, ordered=False, force_drop=False):
 
     elif not dry_run:
         while schemas_with_prefix:
+            recent_errs = []  # Refresh recent_errs at start of while loop
```
Comment (dimitri-yatsenko) [inline datajoint_utilities/dj_search/lists.py]: uninformative comment
```suggestion
            recent_errs = []
```

**Pull request merged**